From 8ddd7c94daf4fbb7196b4be488968f9d6a78827c Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Fri, 6 Oct 2023 08:42:45 +0200
Subject: [PATCH 0603/1141] mtd: spi-nor: add support for Macronix Octal flash

Adding Macronix Octal flash for Octal DTR support.

The octaflash series can be divided into the following types:

MX25 series : Serial NOR Flash.
MX66 series : Serial NOR Flash with stacked die.(Size larger than 1Gb)
LM/UM series : Up to 250MHz clock frequency with both DTR/STR operation.
LW/UW series : Support simultaneous Read-while-Write operation in multiple
               bank architecture. Read-while-write feature which means read
               data one bank while another bank is programing or erasing.

MX25LM : 3.0V Octal I/O
 -https://www.mxic.com.tw/Lists/Datasheet/Attachments/8729/MX25LM51245G,%203V,%20512Mb,%20v1.1.pdf

MX25UM : 1.8V Octal I/O
 -https://www.mxic.com.tw/Lists/Datasheet/Attachments/8967/MX25UM51245G,%201.8V,%20512Mb,%20v1.5.pdf

MX66LM : 3.0V Octal I/O with stacked die
 -https://www.mxic.com.tw/Lists/Datasheet/Attachments/8748/MX66LM1G45G,%203V,%201Gb,%20v1.1.pdf

MX66UM : 1.8V Octal I/O with stacked die
 -https://www.mxic.com.tw/Lists/Datasheet/Attachments/8711/MX66UM1G45G,%201.8V,%201Gb,%20v1.1.pdf

MX25LW : 3.0V Octal I/O with Read-while-Write
MX25UW : 1.8V Octal I/O with Read-while-Write
MX66LW : 3.0V Octal I/O with Read-while-Write and stack die
MX66UW : 1.8V Octal I/O with Read-while-Write and stack die

Change-Id: Ib851fdf0f676321496fa1f4c65beebf366111f83
Signed-off-by: JaimeLiao <jaimeliao@mxic.com.tw>
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/333809
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrice CHOTARD <patrice.chotard@foss.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
---
 drivers/mtd/spi-nor/macronix.c | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/drivers/mtd/spi-nor/macronix.c b/drivers/mtd/spi-nor/macronix.c
index afa79f309223..cacd33d51274 100644
--- a/drivers/mtd/spi-nor/macronix.c
+++ b/drivers/mtd/spi-nor/macronix.c
@@ -118,6 +118,40 @@ static const struct flash_info macronix_nor_parts[] = {
 	{ "mx66u2g45g",	 INFO(0xc2253c, 0, 64 * 1024, 4096)
 		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ)
 		FIXUP_FLAGS(SPI_NOR_4B_OPCODES) },
+	{ "mx66uw2g345gx0", INFO(0xc2943c, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx66lm1g45g", INFO(0xc2853b, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25lm51245g", INFO(0xc2853a, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25lw51245g", INFO(0xc2863a, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25lm25645g", INFO(0xc28539, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx66uw2g345g", INFO(0xc2843c, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx66um1g45g", INFO(0xc2803b, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx66uw1g45g", INFO(0xc2813b, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25uw51345g", INFO(0xc2843a, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25um25645g", INFO(0xc28039, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25uw25645g", INFO(0xc28139, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25um25345g", INFO(0xc28339, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25uw25345g", INFO(0xc28439, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25uw12845g", INFO(0xc28138, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25uw12345g", INFO(0xc28438, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25uw6445g", INFO(0xc28137, 0, 0, 0)
+		PARSE_SFDP },
+	{ "mx25uw6345g", INFO(0xc28437, 0, 0, 0)
+		PARSE_SFDP },
 };
 
 static int macronix_nor_octal_dtr_en(struct spi_nor *nor)
-- 
2.39.2

